### ISSUE TITLE
[Feature] Support additional scrapers (ScreenScraper, TGDB, SteamGridDB)

### ISSUE THREAD

**Issue by XargonWan** [rommapp/romm#212]
Please add [ScreenScraper](https://www.screenscraper.fr/webapi.php) as additional scraper as it seems to be more complete of IGDB on retro gaming.

**Describe the solution you'd like**
There could be a list in the options (similarly to Plex agents) to select which scarapers the user wants, and the config (for an ulimited/ish scraping ScreenScraper needs an user account).
Romm could scrape with the first scraper in the list and then fallback on the other ones if no match (or match but no media) is found.

![image](https://user-images.githubusercontent.com/6771419/235148746-1ff7fbff-9592-446b-8be0-02197ff0b193.png)

Screenscraper is providing more than just cover, but wheel, descriptions, logo and more.

**Describe alternatives you've considered**
I don't know any other scraper services, but maybe someone does.

**Comment by zurdi15** [comment]
Yeah seems like a good idea to integrate more than just IGDB and let the user decide. I need to work a little bit more in the code base to allow more database integrations easily. When that day comes ScreenScraper will be integrated.

**Comment by victor10100111001** [comment]
I think this might be a good choice. Especially since it seems like Zurdi can request an api key that everyone running this project can use. From their website:
```
Who can use the API?
The ScreenScraper API can only be integrated into fully free and distributed applications, or, if not, with the prior authorisation and conditions dictated by the ScreenScraper team. Failure to comply with this rule may be the subject of a breakdown of account, see any legal proceedings.
If you are a developer and want to integrate our API, contact us via the forum to present your software and obtain your usernames and passwords to provide to the API to validate your rights to use it 
```
This would remove some barrier of entry

**Comment by binarygeek119** [comment]
theres also https://thegamesdb.net/ and https://gamesdb.launchbox-app.com/ (althought in not sure is launchbox lets you use their db openly)

**Comment by GAntoine** [comment]
I've used TheGamesDB in a past project and found the library to be fairly extensive. https://github.com/GAntoine/retro-game-names

If we're limited in how many scrapers we can include it gets my vote.

**Comment by GAntoine** [comment]
Alright I've gone through the more prominent DBs/API out there and collected some notes. The ones I think are worth exploring are marked with a ⭐ . Please let me know if I missed any and I'll add them to the list.

### [IGBD](https://api-docs.igdb.com/#getting-started): ~250K games, ~190 platforms ⭐ 

#### Pros
- Free (owned and run by Twitch)
- Extensive list of API endpoints that can get pretty specific
- Great rate limits (4 requests/second limit with no global max)
- Protobuf support

#### Cons
- Greater focus on modern games
- Low number of art assets


### [TheGamesDB](https://api.thegamesdb.net/): ~76K games, ~150 platforms

It seems the _preferred_ way to interact with it is to [mass-download the database](https://forums.thegamesdb.net/viewtopic.php?t=1947) and use the API async to keep your local copy up-to-date.

#### Pros
- Focused on retro games

#### Cons
- API key has to be requested and approved by staff
- Local copy of DB must be synced
- Sparse metadata and related assets


### [MobyGames](https://www.mobygames.com/info/api/): ~330K games, ~320 platforms ⭐ 

#### Pros
- Free for personal use
- Sizeable collection of games and platforms
- Decent collection of crowdsourced assets

#### Cons
- Low API request limits (1 requests/second, max 360 requests/hour)
- API only supports game and platform queries (incl. art assets)

As per the API TOS: "MobyGames must be credited wherever data from the API is used. Typically, a link to the MobyGames page for the game will suffice. "


### [RAWG](https://api.rawg.io/docs/): ~350K games, ~50 platforms ⭐ 

#### Pros

- Free for personal use
- Rich metadata on par with IGDB
- Support for DLCs, achievements, trailers, and more

#### Cons
- Unclear API limits
- Some endpoints locked behind paid plans
- Poor collection of art assets

As per the TOS: "Free for personal use as long as you attribute RAWG as the source of the data and/or images and add an active hyperlink from every page where the data of RAWG is used."


### [GiantBomb](https://www.giantbomb.com/api/documentation/): ~80K games, ~190 platforms

#### Pros

- Decent rate limits (450 requests/15 minutes)
- Extensive API endpoints
- Wiki-style entries for most popular games

#### Cons

- **Paid only (~5$/month)**
- Mediocre collection of art assets


### [ScreenScraper.fr](https://www.screenscraper.fr/webapi2.php?alpha=0&numpage=0): ~300K games, ~105 platforms  ⭐ 

#### Pros

- **Extensive** collection of art assets
- Free for personal use and open-source projects
- Decent API footprint

#### Cons

- Low API rates (20K requests/24 hours for registered users, 3K otherwise)
- French 🇫🇷 🥖 

As per the TOS: "L'API ScreenScraper ne peut être intégré que dans les applications entièrement gratuites et distribuées, ou, dans le cas contraire, avec l'autorisation préalable et les conditions dictées par l'équipe de ScreenScraper."


### [SteamGridDB](https://www.steamgriddb.com/api/v2): ? games, ? platforms

Seems to be mostly focused on art assets based around their concept of "grids", collections of assets uploaded by users.

### Pros

- Prebuilt python library
- Asset collection split by logos, hero images and icons
- ~290K Grids, ~75K heroes, ~75K logos, ~35K icons

### Cons

- Unclear API limits
- No game metadata provided
- More focused on modern games


### [GameTDB](https://www.gametdb.com/): ? games, 7 platforms

Only the following platforms are currently supported: Wii, WiiU, DS, 3DS, Switch, PS3, Gamecube

Pros
- Solid collection of box art assets (for supported platforms)

Cons
- Limited platforms support
- No API/undocumented API

**Comment by Casuallynoted** [comment]
Possibly the ability to import Launchbox's metadata xml files? Not sure how practical that is though

https://forums.launchbox-app.com/topic/52709-launchbox-db-api/

**Comment by maxexcloo** [comment]
Screenscraper is also nice as most frontends (such as ES-DE) use it also so it would be easier to potentially match/share data.

**Comment by LeeTeng2001** [comment]
bump

**Comment by wgoff194** [comment]
I too will bump this, ScreenScraper is absolute gold!

**Comment by adamantike** [comment]
Please only "bump" issues by reacting with 👍, which is useful for developers to sort by most requested features. Any other kind of "+1" comments only generate noise for people subscribed to these issues, looking for actual progress.

**Comment by crispybegs** [comment]
did anything happen with https://screenscraper.fr?
About a month ago I scraped my entire roms collection in batocera with screenscraper and it got 100% matched, but doing the same with igdb in romm over the last 24 hours leaves dozens of gaps that all need to be filled by hand.

**Comment by querylab** [comment]
@crispybegs I have the same problem. I used the SCREENSCRAPER_USER and SCREENSCRAPER_PASSWORD variables shown, but the API didn't recognize them and I didn't get any response. Any idea what could be going wrong?

**Comment by zurdi15** [comment]
> did anything happen with https://screenscraper.fr? About a month ago I scraped my entire roms collection in batocera with screenscraper and it got 100% matched, but doing the same with igdb in romm over the last 24 hours leaves dozens of gaps that all need to be filled by hand.

The Screenscraper integration is currently in the alpha, so expect it for the next release :)

**Comment by zurdi15** [comment]
> [@crispybegs](https://github.com/crispybegs) I have the same problem. I used the SCREENSCRAPER_USER and SCREENSCRAPER_PASSWORD variables shown, but the API didn't recognize them and I didn't get any response. Any idea what could be going wrong?

Which RomM version are you on?

**Comment by crispybegs** [comment]
> > did anything happen with https://screenscraper.fr? About a month ago I scraped my entire roms collection in batocera with screenscraper and it got 100% matched, but doing the same with igdb in romm over the last 24 hours leaves dozens of gaps that all need to be filled by hand.
> 
> The Screenscraper integration is currently in the alpha, so expect it for the next release :)

amazing!

**Comment by gantoine** [comment]
With screenscraper now available in the latest beta (`3.8.0-beta.2`) and primed to be released shortly, I'm content to close this issue as complete! Future work will revolve around an integration with [hasheou](https://github.com/gaseous-project/hasheous).
